shard.go

```diff
@@ -131,6 +149,51 @@ func (s *cacheShard) set(key string, hashedKey uint64, entry []byte) error {
 	}
 }
 
+func (s *cacheShard) setWithoutLock(key string, hashedKey uint64, entry []byte) error {
```

Comment:
Can we use this function in `set`? After release of 1.14 we can loose out rule of no defer

Reply:
@cristaloleg this is not what you wanted right?

Reply:
I suppose @janisz has proposed to use setWithoutLock inside a set. I think we can skip it for a moment and figure out that everything works (in a next PR we can integrate one into another). So if @janisz is agree - we can treat this comment as a resolved one.

Reply:
OK, let's stick with `no defer`. We need to discuss it.